File: live-football-world-cup-score-board/src/ScoreboardService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/scoreboard';

export const startMatch = async (homeTeam, awayTeam) => {
    try {
        await axios.post(`${API_URL}/start`, null, {
            params: { homeTeam, awayTeam }
        });
    } catch (error) {
        return error.response?.data?.message || 'Error starting match';
    }
};

export const updateScore = async (homeTeam, awayTeam, homeScore, awayScore) => {
    try {
        await axios.post(`${API_URL}/update`, null, {
            params: { homeTeam, awayTeam, homeScore, awayScore }
        });
    } catch (error) {
        return error.response?.data?.message || 'Error updating score';
    }
};

export const finishMatch = async (homeTeam, awayTeam) => {
    try {
        await axios.post(`${API_URL}/finish`, null, {
            params: { homeTeam, awayTeam }
        });
    } catch (error) {
        return error.response?.data?.message || 'Error finishing match';
    }
};

export const getSummary = async () => {
    try {
        const response = await axios.get(`${API_URL}/summary`);
        return response.data;
    } catch (error) {
        return 'Error fetching summary';
    }
};

File: src/main/java/org/ozturk/service/ScoreboardService.java
package org.ozturk.service;

import org.ozturk.model.Match;

import java.util.List;

/**
 * The ScoreboardService interface defines the operations supported by a live
 * football World Cup scoreboard service. It allows starting new matches, updating scores,
 * finishing matches, and retrieving a summary of ongoing matches.
 *
 * <p>Matches are uniquely identified by their home and away team names. The scoreboard
 * maintains a list of matches currently in progress.</p>
 *
 * @author Halil Ibrahim Ozturk
 */
public interface ScoreboardService {
    void startMatch(String homeTeam, String awayTeam);

    void updateScore(String homeTeam, String awayTeam, int homeScore, int awayScore);

    void finishMatch(String homeTeam, String awayTeam);

    List<Match> getSummary();
}
